Dockerfile-5_0.ubi8

```diff
@@ -120,7 +120,7 @@ RUN microdnf install --nodocs shadow-utils \
     && useradd -m -d "$CASSANDRA_HOME" -r -g cassandra -G root --uid=999 cassandra \
     && microdnf update && rm -rf /var/cache/yum \
 # Install packages needed during install process
-    && microdnf install --nodocs java-11-openjdk-headless tzdata-java python3 zlib findutils which hostname iproute procps util-linux glibc-langpack-en wget tar \
+    && microdnf install --nodocs java-11-openjdk-headless tzdata-java python3.11 zlib findutils which hostname iproute procps util-linux glibc-langpack-en wget tar \
```

Comment:
question: Is there a reason we enforce 3.11 and not just python3 package? I didn't see such in the description.

Reply:
With the UBI8 image base, when you try to install `python3`, it gets a python 3.6.x version currently. _Some_ versions of `cqlsh` bundled with the various Cassandra bundles warn about Python needing to be at least 3.7, but still seem to allow Python 3.6. Other versions of `cqlsh` warn about python 3.6 and refuse to run. 

I can try this image again to see if it will work with just `python3` and I'll report back.

Reply:
The Cassandra 4.0 UBI8 image is fine with `python3`:

```sh
[root@908c92f0a52e /]# python3 --version
Python 3.6.8
[root@908c92f0a52e /]# cqlsh --version
cqlsh 6.0.0
```

The 5.0 image has the issue if I try to just install `python3`.

```sh
[root@a4005707aad5 /]# python3 --version
Python 3.6.8
[root@a4005707aad5 /]# cqlsh --version
Warning: unsupported version of Python: 3.6
No appropriate Python interpreter found.
```

From that image, if I install python 3.7 or newer:
```sh
[root@5565368e2f59 /]# microdnf install python3.9
```
then remove python3:
```sh
[root@5565368e2f59 /]# microdnf remove python3 python3-pip
```
`cqlsh` works:
```sh
[root@5565368e2f59 /]# python3 --version
Python 3.9.18
[root@5565368e2f59 /]# cqlsh --version
cqlsh 6.2.0
```

So the version of `cqlsh` that comes with the Cassandra bundle determines which version of Python3 it will accept and run with. In the 5.0 image, this is the chunk of code that seems to kick out Python3.6 and lower:
```sh
# test whether a version string matches one of the supported versions for cqlsh
is_supported_version() {
    version=$1
    major_version="${version%.*}"
    minor_version="${version#*.}"
    # python 3.7+ is supported
    if [ "$major_version" = 3 ] && [ "$minor_version" -ge 7 ]; then
        echo "supported"
    else
        echo "unsupported"
    fi
}

run_if_supported_version() {
    # get the interpreter and remove it from argument
    interpreter="$1" shift

    version=$(get_python_version "$interpreter")
    if [ -n "$version" ]; then
        if [ "$(is_supported_version "$version")" = "supported" ]; then
            exec "$interpreter" "$($interpreter -c "import os; print(os.path.dirname(os.path.realpath('$0')))")/cqlsh.py" "$@"
            exit
        else
            echo "Warning: unsupported version of Python:" $version >&2
        fi
    fi
}
```